src/Components/Auth/Login.tsx

```diff
@@ -0,0 +1,222 @@
+  input: {
+    style: {
+      marginBottom: 0
+    }
+  },
+  helpText: {
+    style: {
+      marginBottom: 0
+    }
+  }
+};
+
+const RuleValidate: RuleInterface = {
+  email: {
+    required: 'Địa chỉ email là bắt buộc',
+    pattern: {
+      value: /^[a-z\d._%+-]+@[a-z\d.-]+\.[a-z]{2,4}$/i,
+      message: 'Địa chỉ email không đúng định dạng'
+    }
+  },
+  password: {
+    required: 'Mật khẩu không được trống',
+    minLength: {
+      value: 8,
+      message: 'Mật khẩu phải chứa ít nhất 8 ký tự'
+    },
+    pattern: {
+      value: /^[^ ]*$/i,
+      message: 'Mật khẩu không được chứa dấu khoảng trắng'
+    }
+  }
+};
+
+const FormLogin = styled.div`
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  height: 100%;
+  flex-direction: column;
+`;
+
+function Login() {
+  const {
+    handleSubmit,
+    watch,
+    control,
+    formState: { errors, isValid, isSubmitted }
+  } = useForm<FormValues>();
+
+  const navigate = useNavigate();
+  const dispatch = useAppDispatch();
+  const [requesting, setRequesting] = useState<boolean>(false);
+  const [loginError, setLoginError] = useState<string | null>(null);
+
+  const onSubmit = async (param: FormValues) => {
+    setRequesting(true);
+    const paramLogin: LoginParam = { ...param };
+
+    try {
+      const { user } = await login(paramLogin);
+      setRequesting(false);
+      dispatch(setLogin(user));
+      navigate('/user');
+    } catch (err) {
+      setRequesting(false);
+      if (typeof err == 'string') {
+        setLoginError(err);
+      }
+    }
+  };
+
+  useEffect(() => {
+    const subcription = watch(() => {
+      setLoginError(null);
+    });
+    return () => {
+      subcription.unsubscribe();
+    };
+  }, [watch]);
+
+  return (
+    <FormLogin>
+      <Box
+        sx={{
+          overflow: 'hidden auto',
+          width: '100%',
+          padding: '20px 0',
+          maxHeight: '100vh'
+        }}>
+        <Box
+          sx={{
+            width: '75%',
+            maxWidth: 450,
+            margin: 'auto'
+          }}>
+          <form onSubmit={handleSubmit(onSubmit)}>
+            <Typography variant="h4">Đăng nhập vào tài khoản</Typography>
+            {!!loginError && (
+              <Alert icon={false} severity="error">
+                {loginError}
+              </Alert>
+            )}
+            <Controller
+              control={control}
+              name="email"
+              rules={RuleValidate.email}
+              render={({ field }) => (
+                <TextField
+                  fullWidth
+                  autoFocus
+                  size="small"
+                  margin="dense"
+                  InputProps={propsAttribute.input}
+                  FormHelperTextProps={propsAttribute.helpText}
+                  label="Email (*)"
+                  error={!!errors.email}
+                  helperText={errors.email?.message}
+                  onChange={field.onChange}
+                />
```

Comment:
```suggestion
              render={({ field, fieldState: {error} }) => (
                <TextField
                  fullWidth
                  autoFocus
                  size="small"
                  margin="dense"
                  InputProps={propsAttribute.input}
                  FormHelperTextProps={propsAttribute.helpText}
                  label="Email (*)"
                  error={!!error}
                  helperText={error?.message}
                  onChange={field.onChange}
                />
```

Dùng fieldState có trong hàm render